### ISSUE TITLE
Auto update downloads over metered connections

### ISSUE THREAD

**Issue by Electrenator** [ppy/osu#29749]
### Type

Game behaviour

### Bug description

On windows the game auto updates without checking if someone is on a metered connection. Downloading this could get quite expensive for people sometimes playing while traveling. Noticed it while traveling within a country that I gladly didn't have any data roaming charges for.

Would expect the game to only update manually when the "Metered connection" setting of a WiFi network is enabled.

### Version

2024.817.0 (so might possibly have been fixed by #28743)

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16913779/compressed-logs.zip)
Of which logs from the `1725667215` timestamp and onward are relevant.

**Comment by smallketchup82** [comment]
Please try it with Velopack.

Velopack has much more efficient delta management vs Squirrel. In a nutshell, Velopack will download compressed "patches" for new versions instead of the entirety of the new release. In my experience, these patches are quite small, sometimes just a couple megabytes.

Even though Velopack shouldn't use up too much internet, if you're still interested in completely turning off updates, I can see what I can do (not sure if we'd want that osu-side or velopack-sided)

**Comment by catuhana** [comment]
> Please give Velopack a try
> 
> Velopack has much more efficient [delta management](https://docs.velopack.io/packaging/deltas) vs Squirrel. In a nutshell, Velopack will download highly compressed "patches" for new versions instead of the entirety of a new release. In my experience, these patches are quite small, sometimes just a couple megabytes.
> 
> Update checking shouldn't take too much storage either. Velopack will read the `releases.win.json` from the latest release to determine if it's time to update or not. This file, in the latest release, is only 347 Bytes. It will be doing this every 30 minutes, making for 694 Bytes every hour, making for 16kb total if you keep lazer open for 24 hours nonstop.
> 
> Even though Velopack shouldn't use up too much internet, if you're still interested in completely turning off updates, I can see what I can do. Not sure if we'd want that osu-side or velopack-sided, but I can look into implementing it in a more user friendly manner vs the current environment variable method.

I'd say the issue is not checking the updates, but downloading them. Velopack's delta is already a good step towards smaller and faster updates/update files, but I don't think that fits the purpose of this issue. Roaming data is expensive for some countries (a megabyte is $0.5 in Turkey for most carriers, for example).

I'd also say this could be implemented for downloading maps etc.

**Comment by Electrenator** [comment]
> I'd say the issue is not checking the updates, but downloading them.

Exactly! Mostly try to avoid unexpected data charges as an end user. I love auto updates but try to avoid them when traveling.  
Would nice if downloading the updates could be manually delayed on a metered connection. Allowing someone to search out a place without network charges where they can update while still allowing them to play with counted scores.

> Please give Velopack a try

Definitely am going to use it. Mostly trying to avoid hitting my monthly data limit or unexpected charges when randomly playing on the go.

Though an alternative might be to disable internet myself and fully play offline for those few plays out & about.

**Comment by smallketchup82** [comment]
I'm still not sure what direction to head here. I can get that this might be a pressing issue but wonder how it should be fixed.

From my own thoughts:
- We could add a toggle to settings to control auto updates, but that would be difficult since `VelopackApp` build & run logic is in `osu.Desktop`, where OsuConfigManager likely doesnt exist.
- I can run through the environment variable logic and touch it up (its existence doesnt disable `VelopackApp`  build & run logic which can be a problem)
- Or we can try to detect this ourselves (if the user is on a metered network) and automatically disable auto updating.

I'd like some input from the core team before I proceed with this.

**Comment by peppy** [comment]
The only valid change would be checking the system connection for "low data mode", a user setting on both windows and mac.

No other solution is required or valid.

Also deltas are small and always have been. I feel like this is a complaint without rationale and can be considered low priority.
